### PR TITLE
Sequencer server config

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Extract SequencerServerConfig and make configuration more type-safe

Why should this be merged: 
Improves SequencerServer

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 1]
```diff
@@ -615,4 +609,23 @@ public void shutdown() {
...
+    public static class SequencerServerConfig {
+
+       private final long initialToken;
+       @Default
+       private final long sequencerCacheSize = 250_000;
+
+        public static SequencerServerConfig parse(Map<String, Object> opts) {
+            return SequencerServerConfig.builder()
+                    .initialToken(Utils.parseLong(opts.get("--initial-token")))
+                    .sequencerCacheSize(Utils.parseLong(opts.getOrDefault("--sequencer-cache-size", 250000)))
```
Comment (StrikingThirteen) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: Can you avoid repeating the 250000?

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 2]
```diff
@@ -227,8 +223,7 @@ private String getConflictHashCode(UUID streamId, byte[] conflictParam) {
      * @return Returns the type of token reponse based on whether the txn commits, or the abort
      *     cause.
      */
-    private TokenType txnCanCommit(TxResolutionInfo txInfo, /** Input. */
-                                  AtomicReference<byte[]> conflictKey /** Output. */) {
+    private TokenType txnCanCommit(TxResolutionInfo txInfo, AtomicReference<byte[]> conflictKey) {
```
[the remark this answers is not in the document]
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: Javadoc looks good, these comments seem redundant

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java (hunk 1 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: done

**Comment by Maithem** [comment]
@xnull @StrikingThirteen can you propose some standardization on how configurations are populated and represented to different objects. This and #1623 need to be discussed in more detail.

**Comment by xnull** [comment]
@Maithem the idea is pretty simple, map raw HashMap into a Dto class. 
We have 4 classes LogUnitServer, SequencerServer, LayoutServer, ManagementServer, for all of these servers we can parse config map and map it into ${ClassName}Config, we already have done the work, you can see it in several PR-s including this one.
Not sure if there is anything to discuss now, it is a very simple change it doesn't affect anything, it's completely done we don't use old and new approaches at the same time now.

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 3]
```diff
@@ -554,4 +551,25 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerC
...
         super.shutdown();
     }
+
+    /**
+     * Sequencer server configuration
+     */
+    @Builder
+    @Getter
+    public static class Config {
+        private static final long DEFAULT_CACHE_SIZE = 250000L;
```
Comment (StrikingThirteen) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: 250_000L

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 4]
```diff
@@ -126,26 +128,21 @@
...
-        Map<String, Object> opts = serverContext.getServerConfig();
+        Config config = Config.parse(serverContext.getServerConfig());
+
         this.executor = Executors.newFixedThreadPool(
                 serverContext.getSequencerThreadCount(),
                 new ServerThreadFactory("sequencer-", new ServerThreadFactory.ExceptionHandler())
         );
 
-        long initialToken = Utils.parseLong(opts.get("--initial-token"));
+        long initialToken = config.getInitialToken();
```
Comment (StrikingThirteen) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: Is there any point of letting non address being set in config builder? if not why not moving this logic to config builder?

Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java (hunk 3 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: done

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 5]
```diff
@@ -554,4 +546,31 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg, ChannelHandlerC
...
+    /**
+     * Sequencer server configuration
+     */
+    @Builder
+    @Getter
+    public static class Config {
+        private static final long DEFAULT_CACHE_SIZE = 250_000L;
+
+        private final long initialToken;
+        @Default
```
Comment (vjeko) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: Nit: newline before Default.

**Pull request merged**